### PR TITLE
set gain registers

### PR DESCRIPTION
**Pull request by tharaka27**

### CODE REVIEW

File: AD7173.cpp [hunk 1]
```diff
@@ -275,3 +275,25 @@ int AD7173Class::set_offset_config(adc7173_register_t offset, uint32_t offset_va
 	/* return error code */
 	return 0;
 }
+
+
+int AD7173Class::set_gain_registers(adc7173_register_t gain, uint32_t gain_value) {
+	/* Address Range: 0x38 to 0x3F, Reset: 0x5XXXX0 , Name: GAIN0 to GAIN7 */
+
+	/* add the default gain value */
+	gain_value += 5312784;
```
Comment (CloudyPadmal) [inline AD7173.cpp]: From where is this value come from?

**Comment by CloudyPadmal** [comment]
Hello @tharaka27 , thank you the PR. Please squash the commits into one.

File: AD7173.cpp (hunk 1 above)
Comment (tharaka27) [reply AD7173.cpp]: Sorry for that value. I have updated the code

**Comment by CloudyPadmal** [comment]
Looks good. As you can see this PR has 5 commits. In our projects we have a best practice to squash all commits into one if it's related to a single PR fixing one issue. You may try to squash these commits into one to get a better understanding so that it'll be easier when making new PRs. Thanks.

**Comment by CloudyPadmal** [comment]
Hello @tharaka27 , please have a read at [FOSSASIA best practices](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia/)

Just to summarize things related to this PR, 
1. Always open an issue before working on a feature or a new thing in repo (No issue related to this PR)
2. Make only one PR addressing that issue with one commit (Seems like this PR addressing multiple issues apart from the one mentioned in the title, and 10 commits!)
3. In the PR body, mention what is implemented (No PR body - messages)
4. Once an issue is opened, make sure maintainers or mentors review that issue and give the signal to go ahead. Otherwise it'll be a waste of time if someone opens up an unwanted issue and implements something which then gets closed.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Open issues related to sub features and have multiple PRs to make it easier to review.

**Comment by CloudyPadmal** [comment]
It is highly discouraged closing PRs, especially the ones being reviewed. It would waste all the reviews and comments. Since this is your first PR, please don't do it again. Thanks.

**Comment by tharaka27** [comment]
> It is highly discouraged closing PRs, especially the ones being reviewed. It would waste all the reviews and comments. Since this is your first PR, please don't do it again. Thanks.

Sorry for what happened. Since I am learning by doing things get a mess sometimes. But I'll assure that it won't happen again. And also thank you for your kind reviews. They helped me learn a lot.

**Pull request closed without merging**